[Zhu Tianqi] ip

Tianqi-Zhu opened the pull request:
## DukePro
> Intellij looks so **UGLY**.

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* <del>FAST</del> ***SUPER*** **FAST** to use

All you need to do is
1. download it from [here](https://github.com/Tianqi-Zhu/ip/releases/tag/v0.1)
2. double-click it.
3. add your tasks.
3. let it manage your tasks for you :laughing:

And it is __FREE__!

Features:
 - [x] Managing tasks
 - [x] Managing deadlines
 - [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Tianqi-Zhu committed:
add level 1

Tianqi-Zhu committed:
add level 2

Tianqi-Zhu committed:
add level 3

Tianqi-Zhu committed:
add level 4

Tianqi-Zhu committed:
add text ui test

Tianqi-Zhu committed:
add level 5

Tianqi-Zhu committed:
add level 6

Tianqi-Zhu committed:
add level 7 save load data

Tianqi-Zhu committed:
add level8 datetime conversion

Tianqi-Zhu committed:
resolve merging conflict while merging level 8 branch

Tianqi-Zhu committed:
change structure to be more oop

Tianqi-Zhu committed:
add package

Tianqi-Zhu committed:
add junit test

Tianqi-Zhu committed:
generate jar file

Tianqi-Zhu committed:
add javadoc

Tianqi-Zhu committed:
fix coding style

Tianqi-Zhu committed:
add find

Tianqi-Zhu committed:
fix junit test

Tianqi-Zhu committed:
fix line break

Tianqi-Zhu committed:
add gradle

Tianqi-Zhu committed:
add gui

Tianqi-Zhu committed:
remove binary data file

Tianqi-Zhu committed:
Merge branch 'branch-Level-10'

janjanchen commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,119 @@
+     * @throws DukeExcpetion if the format is not correct
+     */
+    private static LocalTime parseTime(String timeString) throws DukeExcpetion {
+        String expanded = timeString.substring(0, 2) + ":" + timeString.substring(2) + ":00";
```
Perhaps the variable name `expanded` could be changed to make it more descriptive? (e.g. expandedTimeString)

janjanchen commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,53 @@
+        this.isTimeString = false; 
+    }
+
+    private String getTime() {
```
Perhaps `getTime()` could be named as `getDateAndTime()` since the method will always return date, while time is only included if the user added more details to `Event`

sw-boi reviewed (commented):
LGTM! Aside from some nitpicks mentioned, your code looks good.

DonnyS57 commented on src/main/java/duke/DukeExcpetion.java:
```diff
@@ -0,0 +1,14 @@
+/**
+ * Exception when users type invalid input.
+ */
+public class DukeExcpetion extends Exception {
```
This is something minor but Is it better if the Class is named DukeException? I think it will increase the readability of the code!

DonnyS57 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,119 @@
+     * @return Commands to execute or tasks to put into list.
+     * @throws DukeExcpetion If the input is valid.
+     */
+    public static Executable parse(String newInput, TaskList taskList) throws DukeExcpetion {
```
Perhaps the if-else conditions can be separated into different methods? I think this will greatly increase the readability of your code as the parse method is currently quite verbose.

DonnyS57 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,130 @@
+     * 
+     * @param index Index of the task to be deleted. 
+     */
+    public String deteteTask(int index) {
```
Perhaps the name can be changed to deleteTask? (Another very minor change that can enhance readability).

DonnyS57 reviewed (commented):
Overall, I think most of the code adheres to the Coding Standard aside from the minor problems that I highlighted. Keep up the good work!

Tianqi-Zhu committed:
improve code quality
Divided duke package into smaller subpackages.

Divide parse method into smaller methods to avoid long deep nested code.

Tianqi-Zhu committed:
improve code quality
removed obselete command line functions, added greeting for gui and chaged io exception to assertion.

Tianqi-Zhu committed:
add assertion

Tianqi-Zhu committed:
Merge pull request #2 from Tianqi-Zhu/A-Assertions

add assertion
When there is ioexception when loading and storing, it 
uses assertion to terminate the programming instead of 
printing stack trace of error message in command line like
before.

Tianqi-Zhu committed:
resolve merege conflict with master

Tianqi-Zhu committed:
Merge pull request #1 from Tianqi-Zhu/branch-A-CodeQuality

Branch a code quality

Tianqi-Zhu committed:
add doafter
It enables duke to record tasks to be donf after a particular time.

Tianqi-Zhu committed:
Merge branch 'master' of github.com:Tianqi-Zhu/ip

Tianqi-Zhu committed:
add jar v0.2

Tianqi-Zhu committed:
add user guide

Tianqi-Zhu committed:
add ui image

Tianqi-Zhu committed:
fix md space issue

Tianqi-Zhu committed:
fix line spacing

Tianqi-Zhu committed:
improve line spacing

Tianqi-Zhu committed:
fix command style issues

Tianqi-Zhu committed:
update user guide exit

Tianqi-Zhu committed:
improve junit test

Tianqi-Zhu committed:
add final jar

Tianqi-Zhu committed:
fix bugs and release

Tianqi-Zhu committed:
Update README.md

Tianqi-Zhu committed:
fix header comments style
